tests/integration/cli/test_cast.py

```
@@ -0,0 +1,294 @@
+        assert "cast failed: " in capsys.readouterr().err
+
+
+class TestCastId:
+    @staticmethod
+    def test_two_casts_of_one_ritual_get_distinct_ids(tmp_path, monkeypatch):
+        (tmp_path / "rituals.py").write_text(_RITUALS)
+        monkeypatch.chdir(tmp_path)
+        seen: list[str] = []
+        real = _inits.Grimoire
+
+        def _record(*, cast_id: str, **kwargs):
```

Ruff (0.15.21)

[warning] Missing return type annotation for private function `_record`

(ANN202)